### PR TITLE
Query params in config commands

### LINKED ISSUE #4934: Error when trying to set a global variable using a query parameter
- EdgeDB Version: 2.9+4fbadd2
- EdgeDB CLI Version: EdgeDB CLI 2.2.6+7eabbf9
- OS Version: MacOS Monterey 12.5.1

The docs are ambiguous about it, but if this is not a bug, you may read it as a feature request: it should be possible to set a global variable using a query parameter. Right now it isn't, as it causes EdgeDB to throw an error. 

Steps to Reproduce:

1. With a global variable defined in the schema, try to set it in the shell using a query parameter. Assuming dynamic 

```
create global my_var -> str;
set global my_var := <str>$my_param;
```
results in:

```shell
edgedb> create global my_var -> str;
....... set global my_var := <str>$my_param;
OK: CREATE GLOBAL
edgedb error: InternalServerError: my_param
  Hint: This is most likely a bug in EdgeDB. Please consider opening an issue ticket at https://github.com/edgedb/edgedb/issues/new?template=bug_report.md
  Server traceback:
      Traceback (most recent call last):
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/pgsql/compiler/__init__.py", line 120, in compile_ir_to_sql_tree
          qtree = dispatch.compile(ir_expr, ctx=ctx)
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/functools.py", line 889, in wrapper
          return dispatch(args[0].__class__)(*args, **kw)
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/pgsql/compiler/config.py", line 47, in compile_ConfigSet
          val = _compile_config_value(op, ctx=ctx)
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/pgsql/compiler/config.py", line 505, in _compile_config_value
          val = dispatch.compile(expr, ctx=subctx)
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/functools.py", line 889, in wrapper
          return dispatch(args[0].__class__)(*args, **kw)
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/pgsql/compiler/expr.py", line 60, in compile_Set
          _compile_set_impl(ir_set, ctx=ctx)
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/pgsql/compiler/expr.py", line 114, in _compile_set_impl
          _compile_set(ir_set, ctx=ctx)
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/pgsql/compiler/expr.py", line 677, in _compile_set
          relgen.get_set_rvar(ir_set, ctx=ctx)
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/pgsql/compiler/relgen.py", line 162, in get_set_rvar
          return _process_toplevel_query(ir_set, ctx=ctx)
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/pgsql/compiler/relgen.py", line 265, in _process_toplevel_query
          rvars = _get_set_rvar(ir_set, ctx=ctx)
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/pgsql/compiler/relgen.py", line 418, in _get_set_rvar
          rvars = process_set_as_expr(ir_set, stmt, ctx=ctx)
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/pgsql/compiler/relgen.py", line 2175, in process_set_as_expr
          set_expr = dispatch.compile(ir_set.expr, ctx=newctx)
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/functools.py", line 889, in wrapper
          return dispatch(args[0].__class__)(*args, **kw)
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/pgsql/compiler/expr.py", line 131, in compile_Parameter
          index = ctx.argmap[expr.name].index
      KeyError: 'my_param'
      
      The above exception was the direct cause of the following exception:
      
      Traceback (most recent call last):
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/server/compiler_pool/worker_proc.py", line 55, in worker
          res = meth(*args)
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/server/compiler_pool/worker.py", line 160, in compile
          units, cstate = COMPILER.compile(
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/server/compiler/compiler.py", line 2277, in compile
          unit_group = self._compile(ctx=ctx, source=source)
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/server/compiler/compiler.py", line 1807, in _compile
          return self._try_compile(ctx=ctx, source=source)
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/server/compiler/compiler.py", line 1871, in _try_compile
          comp, capabilities = self._compile_dispatch_ql(
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/server/compiler/compiler.py", line 1776, in _compile_dispatch_ql
          self._compile_ql_config_op(ctx, ql),
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/server/compiler/compiler.py", line 1668, in _compile_ql_config_op
          sql_text, _ = pg_compiler.compile_ir_to_sql(
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/pgsql/compiler/__init__.py", line 152, in compile_ir_to_sql
          qtree = compile_ir_to_sql_tree(
        File "/Users/********/Library/Application Support/edgedb/portable/2.9/lib/python3.10/site-packages/edb/pgsql/compiler/__init__.py", line 135, in compile_ir_to_sql_tree
          raise errors.InternalServerError(*args) from e
      edb.errors.InternalServerError: my_param
edgedb> 

  ```

### PR DESCRIPTION
**Pull request by aljazerzen**
Closes #4934

### CODE REVIEW

File: edb/edgeql/compiler/stmtctx.py [hunk 1]
```diff
@@ -214,10 +214,15 @@ def fini_expression(
...
 
+    # Collect query parameters
+    params = collect_params(ctx)
+
     # ConfigSet and ConfigReset don't like being part of a Set, so bail early
     if isinstance(ir.expr, (irast.ConfigSet, irast.ConfigReset)):
         ir.expr.scope_tree = ctx.path_scope
         ir.expr.globals = list(ctx.env.query_globals.values())
+        ir.expr.params = params
+        ir.expr.schema = ctx.env.schema
```
Comment (aljazerzen) [inline edb/edgeql/compiler/stmtctx.py:225]: @msullivan from the top of your head, is there an obvious blocker that prevents ConfigCommand be a part of Statement?

If you don't think of one, I'll try to fit it in (in a following PR)

File: tests/test_edgeql_globals.py [hunk 2]
```diff
@@ -404,6 +404,20 @@ async def test_edgeql_globals_12(self):
             [16]
         )
 
+    async def test_edgeql_globals_13(self):
```
Comment (msullivan) [inline tests/test_edgeql_globals.py:407]: Could we add a test with config commands too? I seem to recall we might try to use the static evaluator on those so it's worth seeing what happens

File: edb/pgsql/compiler/__init__.py [hunk 3]
```diff
@@ -65,7 +65,7 @@ def compile_ir_to_sql_tree(
         ]
     ] = None,
     backend_runtime_params: Optional[pgparams.BackendRuntimeParams]=None,
-) -> Tuple[pgast.Base, context.Environment]:
```
Comment (msullivan) [inline edb/pgsql/compiler/__init__.py:68]: Maybe these functions ought to return a namedtuple or a dataclass. Doesn't need to happen now though.

**Review by msullivan — APPROVED** [review]
This is great!

File: tests/test_edgeql_globals.py (hunk 2 above)
Comment (aljazerzen) [reply tests/test_edgeql_globals.py:407]: I'm not sure if this is possible. The non-global "config set" commands we have are:
- `set module mod;`
- `set alias al as module mod;`

https://www.edgedb.com/docs/reference/edgeql/sess_set_alias#set

Comment (msullivan) [reply tests/test_edgeql_globals.py:407]: @aljazerzen https://www.edgedb.com/docs/reference/admin/configure#configure

**Pull request merged**